lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -4153,87 +3965,1104 @@ inline llvm::raw_ostream &operator<<(llvm::raw_ostream &os,
+    // result.
+    SmallVector<SILParameterInfo, 8> dfParams;
+    SmallVector<SILResultInfo, 8> dfResults;
+    auto origParams = origTy->getParameters();
+    auto indices = attr->getIndices();
+
+    // Add differential results.
+    auto origResInfo = origTy->getResults()[indices.source];
+    dfResults.push_back(
+        SILResultInfo(origResInfo.getType()
+                          ->getAutoDiffAssociatedTangentSpace(lookupConformance)
+                          ->getCanonicalType(),
+                      origResInfo.getConvention()));
+
+    // Add differential parameters for the requested wrt parameters.
+    for (auto i : indices.parameters->getIndices()) {
+      auto origParam = origParams[i];
+      dfParams.push_back(SILParameterInfo(
+          origParam.getType()
+              ->getAutoDiffAssociatedTangentSpace(lookupConformance)
+              ->getCanonicalType(),
+          origParam.getConvention()));
+    }
+
+    // Accept a differential struct in the differential parameter list. This is
+    // the returned differential's closure context.
+    auto *origEntry = original->getEntryBlock();
+    auto *dfStruct = linearMapInfo->getLinearMapStruct(origEntry);
+    auto dfStructType =
+        dfStruct->getDeclaredInterfaceType()->getCanonicalType();
+    dfParams.push_back({dfStructType, ParameterConvention::Direct_Owned});
+
+    Mangle::ASTMangler mangler;
+    auto diffName = original->getASTContext().getIdentifier(
+        mangler.mangleAutoDiffLinearMapHelper(
+            original->getName(), AutoDiffLinearMapKind::Differential,
+            indices)).str();
+    auto diffGenericSig = getAssociatedFunctionGenericSignature(attr, original);
+    auto *diffGenericEnv =
+        diffGenericSig ? diffGenericSig->createGenericEnvironment() : nullptr;
+    auto diffType = SILFunctionType::get(
+        diffGenericSig, origTy->getExtInfo(), origTy->getCoroutineKind(),
+        origTy->getCalleeConvention(), dfParams, {}, dfResults, None,
+        original->getASTContext());
+
+    SILOptFunctionBuilder fb(context.getTransform());
+    // The generated tangent linkage is set to Hidden because generated tangent
+    // are never called cross-module.
+    auto linkage = SILLinkage::Hidden;
+    auto *differential = fb.createFunction(
+        linkage, diffName, diffType, diffGenericEnv, original->getLocation(),
+        original->isBare(), IsNotTransparent, original->isSerialized(),
+        original->isDynamicallyReplaceable());
+    differential->setDebugScope(
+        new (module) SILDebugScope(original->getLocation(), differential));
+
+    return differential;
+  }
+
+  /// Initial function call to set up the differential function. This includes:
+  /// - Adding all the differential blocks.
+  /// - Adding all the arguments to the function's entry basic block.
+  /// - Adding the tangent values of the parameters.
+  /// - Verification checks like varied result.
```

Comment:
- It's not a verification -- it's a check for emitting necessary warnings. "Verification" usually means checking whether you have invalid code. However, unvaried result is not invalid code.
- Describe what it does accurately.
```suggestion
  /// - Checking for unvaried result and emitting related warnings.
```